openapi-cli/src/test/java/io/ballerina/openapi/generators/openapi/EnumTypeTests.java

```diff
@@ -60,6 +60,20 @@ public void testEnumWithConstantValue() throws IOException {
         TestUtils.compareWithGeneratedFile(ballerinaFilePath, "data_type/enum_with_value.yaml");
     }
 
+    @Test(description = "When the query and path has reference to enum with constant value")
+    public void testEnumWithQueryParameter() throws IOException {
+        Path ballerinaFilePath = RES_DIR.resolve("data_type/enum_query.bal");
+        //Compare generated yaml file with expected yaml content
+        TestUtils.compareWithGeneratedFile(ballerinaFilePath, "data_type/enum_query.yaml");
+    }
+
+    @Test(description = "When the header has reference to enum with constant value")
+    public void testEnumWithHeaderParameter() throws IOException {
+        Path ballerinaFilePath = RES_DIR.resolve("data_type/enum_header.bal");
+        //Compare generated yaml file with expected yaml content
+        TestUtils.compareWithGeneratedFile(ballerinaFilePath, "data_type/enum_header.yaml");
+    }
+
```

Comment:
@lnash94 Have we added any negative tests related to unsupported enums scenarios?

Reply:
Currently, resources functions support enum types in all the parameters: path, query, header, payload and return type. These all are covered here. Therefore wondering to find any unsupported scenario for an enum within the resource function.